common/js/src/executable-module/executable-module.js

```diff
@@ -0,0 +1,77 @@
+/** @license
+ * Copyright 2020 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview This file contains an abstract interface that can be used for
+ * loading and interacting with a binary executable module (e.g., an
+ * Emscripten/WebAssembly module).
+ */
+
+goog.provide('GoogleSmartCard.ExecutableModule');
+
+goog.require('goog.log.Logger');
+goog.require('goog.messaging.AbstractChannel');
+goog.require('goog.Disposable');
+goog.require('goog.Promise');
+
+goog.scope(function() {
+
+const GSC = GoogleSmartCard;
+
+/**
+ * An interface that can be used for loading and interacting with a binary
+ * executable module.
+ * The object gets disposed of when the module fails to load or crashes later.
+ * @abstract @constructor
+ * @extends goog.Disposable
+ */
+GSC.ExecutableModule = function() {};
```

Comment:
It seems that classes should be defined with the `class` keyword (ES6 style) or with `goog.defineClass`: https://google.github.io/styleguide/jsguide.html#features-classes-old-style
https://www.w3schools.com/js/js_es6.asp#mark_class

Reply:
Ack - Same as above, the migration to the new syntax will be a separate effort tracked by #264.